File: python/spaces/lookup.py
"""
Spaces Lookup - X API v2
========================
Endpoint: GET https://api.x.com/2/spaces
Docs: https://developer.x.com/en/docs/twitter-api/spaces/lookup/api-reference/get-spaces

Authentication: Bearer Token (App-only) or OAuth (User Context)
Required env vars: BEARER_TOKEN
"""

import requests
import os
import json

bearer_token = os.environ.get("BEARER_TOKEN")


def create_url():
    # Space IDs to look up (comma-separated)
    space_ids = "ids=1DXxyRYNejbKM"
    
    # Space fields are adjustable. Options include:
    # host_ids, created_at, creator_id, id, lang, invited_user_ids,
    # participant_count, speaker_ids, started_at, ended_at, subscriber_count,
    # topic_ids, state, title, updated_at, scheduled_start, is_ticketed
    space_fields = "space.fields=host_ids,created_at,creator_id,participant_count,title,state"
    url = "https://api.x.com/2/spaces?{}&{}".format(space_ids, space_fields)
    return url


def bearer_oauth(r):
    """
    Method required by bearer token authentication.
    """
    r.headers["Authorization"] = f"Bearer {bearer_token}"
    r.headers["User-Agent"] = "v2SpacesLookupPython"
    return r


def connect_to_endpoint(url):
    response = requests.request("GET", url, auth=bearer_oauth)
    print(response.status_code)
    if response.status_code != 200:
        raise Exception(response.status_code, response.text)
    return response.json()


def main():
    url = create_url()
    json_response = connect_to_endpoint(url)
    print(json.dumps(json_response, indent=4, sort_keys=True))


if __name__ == "__main__":
    main()

File: javascript/spaces/lookup.js
/**
 * Spaces Lookup - X API v2
 * 
 * Endpoint: GET https://api.x.com/2/spaces
 * Docs: https://developer.x.com/en/docs/twitter-api/spaces/lookup/api-reference/get-spaces
 * 
 * Authentication: Bearer Token (App-only) or OAuth (User Context)
 * Required env vars: BEARER_TOKEN
 */

const needle = require('needle');

const token = process.env.BEARER_TOKEN;
const endpointURL = "https://api.x.com/2/spaces";

async function getRequest() {
    const params = {
        "ids": "1DXxyRYNejbKM",
        "space.fields": "host_ids,created_at,creator_id,participant_count,title,state"
    };

    const res = await needle('get', endpointURL, params, {
        headers: {
            "User-Agent": "v2SpacesLookupJS",
            "authorization": `Bearer ${token}`
        }
    });

    if (res.body) {
        return res.body;
    } else {
        throw new Error('Unsuccessful request');
    }
}

(async () => {
    try {
        const response = await getRequest();
        console.dir(response, { depth: null });
    } catch (e) {
        console.log(e);
        process.exit(-1);
    }
    process.exit();
})();
